laws/src/main/scala/cats/laws/discipline/NonEmptyTraverseTests.scala

```diff
@@ -0,0 +1,59 @@
+package cats.laws.discipline
+
+
+import org.scalacheck.{Arbitrary, Cogen, Prop}
+import Prop.forAll
+import cats.{Applicative, Eq, Monoid, NonEmptyTraverse}
+import cats.laws.NonEmptyTraverseLaws
+
+
+trait NonEmptyTraverseTests[F[_]] extends TraverseTests[F] with ReducibleTests[F] {
+  def laws: NonEmptyTraverseLaws[F]
+
+  def nonEmptyTraverse[G[_]: Applicative, A: Arbitrary, B: Arbitrary, C: Arbitrary, M: Arbitrary, X[_]: Applicative, Y[_]: Applicative](implicit
+    ArbFA: Arbitrary[F[A]],
+    ArbXB: Arbitrary[X[B]],
+    ArbYB: Arbitrary[Y[B]],
+    ArbYC: Arbitrary[Y[C]],
+    ArbFB: Arbitrary[F[B]],
+    ArbFGA: Arbitrary[F[G[A]]],
+    ArbGB: Arbitrary[G[B]],
+    CogenA: Cogen[A],
+    CogenB: Cogen[B],
+    CogenC: Cogen[C],
+    CogenM: Cogen[M],
+    M: Monoid[M],
+    MB: Monoid[B],
+    EqFA: Eq[F[A]],
+    EqFC: Eq[F[C]],
+    EqG: Eq[G[Unit]],
+    EqM: Eq[M],
+    EqA: Eq[A],
+    EqB: Eq[B],
+    EqXYFC: Eq[X[Y[F[C]]]],
+    EqXFB: Eq[X[F[B]]],
+    EqYFB: Eq[Y[F[B]]],
+    EqOptionA: Eq[Option[A]]
+  ): RuleSet = {
+    implicit def EqXFBYFB : Eq[(X[F[B]], Y[F[B]])] = new Eq[(X[F[B]], Y[F[B]])] {
+      override def eqv(x: (X[F[B]], Y[F[B]]), y: (X[F[B]], Y[F[B]])): Boolean =
+        EqXFB.eqv(x._1, y._1) && EqYFB.eqv(x._2, y._2)
+    }
+    new RuleSet {
+      def name: String = "nonEmptyTraverse"
+      def bases: Seq[(String, RuleSet)] = Nil
+      def parents: Seq[RuleSet] = Seq(traverse[A, B, C, M, X, Y], reducible[G, A, B])
+      def props: Seq[(String, Prop)] = Seq(
+        "nonEmptyTraverse identity" -> forAll(laws.nonEmptyTraverseIdentity[A, C] _),
+        "nonEmptyTraverse sequential composition" -> forAll(laws.traverseSequentialComposition[A, B, C, X, Y] _),
+        "nonEmptyTraverse parallel composition" -> forAll(laws.traverseParallelComposition[A, B, X, Y] _),
```

Comment:
should we add tests for  `laws.nonEmptyTraverseParallelComposition` and `laws.nonEmptyTraverseSequentialComposition` ?